reflex-dom/src/Reflex/Dom/Internal.hs

```diff
@@ -20,6 +21,19 @@ import qualified Reflex.Dom.Main as Main
 #if defined(ghcjs_HOST_OS)
 run :: a -> a
 run = id
+#elif defined(MIN_VERSION_jsaddle_warp)
+import Data.Maybe (fromMaybe)
+import Data.Monoid ((<>))
+import Language.Javascript.JSaddle (JSM)
+import qualified Language.Javascript.JSaddle.Warp as JW
+import System.Environment (lookupEnv)
+
+run :: JSM () -> IO()
+run jsm = do
+  port :: Int <- read <$> getEnvOr "JSADDLE_WARP_PORT" "3003"
+  putStrLn $ "Running jsaddle-warp server in port " <> show port
+  JW.run port jsm
+    where getEnvOr e v = lookupEnv e >>= return . fromMaybe v
```

Comment:
`fromMaybe v <$> lookupEnv e` might be cleaner

Reply:
Also, using a `where` clause like this is a bit confusing, since it's visually nested under the `do` block.  I wonder if it wouldn't be better to simply rewrite line 33 to say:
`port <- maybe 3003 read <$> lookupEnv "JSADDLE_WARP_PORT"`

Reply:
:+1:  That's a good place to consider `maybe` in place of `fromMaybe`.